fed/proxy/barriers.py

```diff
@@ -67,94 +56,66 @@ def pop_from_two_dim_dict(the_dict, key_a, key_b):
     return the_dict[key_a].pop(key_b)
 
 
-class SendDataService(fed_pb2_grpc.GrpcServiceServicer):
-    def __init__(self, all_events, all_data, party, lock):
-        self._events = all_events
-        self._all_data = all_data
+class SendProxy(abc.ABC):
```

Comment:
At high level, I strongly suggest that we need to move these abstract proxy classes to a new file and that file doesn't contain any implementation detail. Maybe call it `base_proxy.py` or anything like this.

Reply:
Good suggestion ~